trap/src/main/java/com/cursorinsight/trap/TrapManager.kt

```diff
@@ -299,15 +303,7 @@ class TrapManager internal constructor(
      */
     private fun haltAll(activity: Activity) {
         try {
-            if (isRunning) {
-                isRunning = false
-                for (collector in collectors.values) {
-                    collector.stop(activity)
-                }
-                buffer.add(stopMessage())
-                reporter.stop()
-                inLowDataMode = null
```

Comment:
New `stopReporterAndCollectors()` function does not have this assignment. Did you intentionally omit it?

Reply:
You, are right, I've added it back to runAll() because it is more meaningful there.

When the data collection is restarted with runAll we would like to ensure that it is started based on the current network status. So it is set to null, and will get a non null value and start the data collection, when the network callback is called immediately after the subscription is done in subscribeOnNotifications.